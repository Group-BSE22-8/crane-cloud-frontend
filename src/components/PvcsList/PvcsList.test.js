/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import PvcsListPage, { mapStateToProps } from "./";

const PvcsListPageProps = {
  clusters: { log1: 1, log2: 2 },
  match: { params: { projectId: "1" } },
  getPvcs: jest.fn(),
};
// {} ={}
describe("Testing the PVCs Page component", () => {
  const WrapperPvcsListPage = PvcsListPage.WrappedComponent;
  const PvcsListPageComponent = shallow(
    <WrapperPvcsListPage {...PvcsListPageProps} />
  );
  it("should match the snapshot for PvcsListPage after adding props", () => {
    PvcsListPageComponent.setProps(PvcsListPageProps);
    expect(PvcsListPageComponent).toBeDefined();
  });
  it("matchs the PvcsListPage component snapshot", () => {
    expect(PvcsListPageComponent).toMatchSnapshot();
  });
});

describe("Testing the exported mapstate to props and dispatch", () => {
  it("matches the mapstostate", () => {
    expect(
      mapStateToProps({
        pvcsReducer: {
          isFetched: false,
          pvcs: [],
          isRetrieving: false,
        },
      })
    ).toEqual({
      isFetched: false,
      pvcs: [],
      isRetrieving: false,
    });
  });
});
