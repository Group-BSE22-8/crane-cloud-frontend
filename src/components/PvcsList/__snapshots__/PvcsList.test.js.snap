// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the PVCs Page component matchs the PvcsListPage component snapshot 1`] = `ShallowWrapper {}`;
